python/cudf/cudf/core/series.py

```diff
@@ -1527,14 +1517,7 @@ def replace(self, to_replace, replacement):
         if not is_scalar(to_replace):
             if is_scalar(replacement):
                 all_nan = replacement is None
-                if all_nan:
```

Comment:
The other approach is moving the numeric handling logic out of `find_and_replace` and out to the method here.